### PR TITLE
Allocation diagram

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes https://github.com/Deltares/Ribasim/issues/761.

### CODE REVIEW

File: docs/core/allocation.qmd [hunk 1]
```diff
@@ -6,6 +6,34 @@ Allocation is the process of assigning an allocated abstraction flow rate to use
 
 The allocation problem is solved per subnetwork of the Ribasim model, from which a new graph is created called the allocation graph. The allocation graph is used to formulate an optimization problem with the [JuMP](https://jump.dev/JuMP.jl/stable/) package, which is solved using the [HiGHS solver](https://highs.dev/). For more information see also the example of solving the maximum flow problem with `JuMP.jl` [here](https://jump.dev/JuMP.jl/stable/tutorials/linear/network_flows/#The-max-flow-problem).
 
+```{mermaid}
+sequenceDiagram
+  participant Sim as Process: Ribasim simulation
+  participant Param as Data: Parameters
+  participant State as Data: State
+  participant Optim as process: Allocation optimization
```
Comment (visr) [inline docs/core/allocation.qmd:14]:
```suggestion
  participant Optim as Process: Allocation optimization
```
Capitalize Process like the others.

File: docs/core/allocation.qmd [hunk 2]
```diff
@@ -6,6 +6,34 @@ Allocation is the process of assigning an allocated abstraction flow rate to use
...
+    Sim->>Sim: Time integration step
+    Sim-->>State: Update state
+    deactivate Sim
+  end
+  loop Allocation optimization, per allocation network
+    Param-->>Optim: Input
+    activate Optim
+    State-->>Optim: Input
+    Optim->>Optim: Optimize basin allocations if below target level
+    Optim->>Optim: Optimize user allocation, per priority
```
Comment (visr) [inline docs/core/allocation.qmd:31]:
```suggestion
    Optim->>Optim: Optimize Basin allocations if below target level
    Optim->>Optim: Optimize User allocations, per priority
```
Capitalized to make it clear we are talking about specific node types.

File: docs/core/allocation.qmd [hunk 3]
```diff
@@ -6,6 +6,34 @@ Allocation is the process of assigning an allocated abstraction flow rate to use
 
 The allocation problem is solved per subnetwork of the Ribasim model, from which a new graph is created called the allocation graph. The allocation graph is used to formulate an optimization problem with the [JuMP](https://jump.dev/JuMP.jl/stable/) package, which is solved using the [HiGHS solver](https://highs.dev/). For more information see also the example of solving the maximum flow problem with `JuMP.jl` [here](https://jump.dev/JuMP.jl/stable/tutorials/linear/network_flows/#The-max-flow-problem).
 
+```{mermaid}
```
Comment (visr) [inline docs/core/allocation.qmd:9]: The Mermaid source indicates that this is a sequence diagram, but that term doesn't appear in the output. Could you briefly explain in the text above that this is a sequence diagram with a link to for instance https://en.wikipedia.org/wiki/Sequence_diagram? That could help for people that don't know how to read it, like me 5 minutes ago.

**Review by visr — COMMENTED** [review]
This is really nice to have! The diagram matches my understanding as well, left a few minor comments.

**Comment by deltamarnix** [comment]
Hey that's a good start! I would like to see in the image maybe:

* Are these two loops themselves also looped? Or is this really one time only?
* How is the allocation process started? There must be some starting point in the code.
* Are the Ribasim process and the allocation process really two different executables/processes. Or are we talking about modules within the same application?

Good use of dashed arrows and activity bars!

**Comment by SouthEndMusic** [comment]
@deltamarnix go wild :)

**Comment by SouthEndMusic** [comment]
Updated version:

sim_and_opt_diagram

**Comment by SouthEndMusic** [comment]
One could argue that the latest version above contains to much detail for where it is in the docs. Maybe this version could be in the developer docs and a stripped version where this one is now

**Review by visr — APPROVED** [review]

**Pull request merged**